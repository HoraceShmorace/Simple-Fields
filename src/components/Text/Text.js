import { useState } from 'react'
import { bool, func, number, string } from 'prop-types'
import Fieldset from '../Fieldset/Fieldset'
import styles from './Text.module.css'

const Text = ({ description, id, label, maxLength, onChange, placeholder, isRequired, type, value: defaultValue }) => {
  const [value, setValue] = useState(defaultValue)
  const handleChange = e => {
    const { id, value } = e.target.value
    
    setValue(value)

    if (typeof onChange === 'function') onChange({ [id]: value })
  }
  return (
    <Fieldset className={styles.root} description={description} label={label} htmlFor={id}>
     <input maxLength={maxLength} onChange={handleChange} placeholder={placeholder} required={isRequired} type={type} defaultValue={value} />
    </Fieldset>
  )
}

Text.propTypes = {
  description: string,
  id: string.isRequired,
  label: string,
  maxLength: number,
  onChange: func,
  placeholder: string,
  isRequired: bool,
  type: string,
  value: string
}

Text.defaultProps = {
  description: null,
  maxLength: 1028,
  onChange: null,
  placeholder: null,
  isRequired: null,
  type: 'text',
  value: ''
}

export default Text
