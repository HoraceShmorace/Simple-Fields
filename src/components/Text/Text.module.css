input:invalid {
  border-color: red;
}

input:valid {
  border-color: black;
}

.root input[type="text"] {
  background-color: #f5f5f5;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #444;
  font-family: helvetica;
  font-size: 1.75rem;
  font-weight: bold;
  margin-left: -2px;
  outline-width: 0;
  padding: 6px;
  width: calc(100% - 12px);
}

.root input[type="text"]::placeholder {
  color: #aaa;
  font-size: 1.5rem;
}

