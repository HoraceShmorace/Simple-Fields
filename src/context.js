import { createContext, useReducer } from 'react'
import { node } from 'prop-types'
const ADD_DATA_ACTION = 'ADD_DATA'
const RESET_DATA_ACTION = 'RESET_DATA'
const initialState = {}
const reducer = (state, { payload, type }) => {
  switch (type) {
    case ADD_DATA_ACTION:
      return {
        ...state,
        ...payload
      }
    case RESET_DATA_ACTION:
      return {}
    default:
      return state
  }
}

export const Context = createContext(initialState)

export const ContextProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)
  const addData = payload => dispatch({ payload, type: ADD_DATA_ACTION })
  const resetData = () => dispatch({ payload: initialState, type: RESET_DATA_ACTION })

  return (
    <Context.Provider value={{ addData, resetData, state }}>
      {children}
    </Context.Provider>
  )
}

ContextProvider.propTypes = {
  children: node.isRequired
}
