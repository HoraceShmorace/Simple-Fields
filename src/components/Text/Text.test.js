import { render } from 'enzyme'
import toJson from 'enzyme-to-json'
import Text from './Text'
import { ContextProvider } from '~src/context'
import mockData from '~mock/form.json'

const FIELD_TYPE_TEXT = 'text'
const { fields, id } = mockData

describe('Text', () => {
  describe('when rendered', () => {
    const props = fields.find(({ type }) => type === FIELD_TYPE_TEXT)
    const component = render(
      <ContextProvider formId={id}>
        <Text {...props} />
      </ContextProvider>
    )

    it('should match snapshot', () => {
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
