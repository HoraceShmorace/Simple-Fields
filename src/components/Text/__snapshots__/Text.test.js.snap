// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text when rendered should match snapshot 1`] = `
<input
  aria-required="false"
  id="115390410"
  name="115390410"
  placeholder=""
  type="text"
  value=""
/>
`;
